src/components/Pressable/PressableProps.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  ColorValue,
+  AccessibilityProps,
+  ViewProps,
+  Insets,
+  StyleProp,
+  ViewStyle,
+} from 'react-native';
+
+export interface PressableStateCallbackType {
+  readonly pressed: boolean;
+}
+
+export interface PressableAndroidRippleConfig {
+  color?: null | ColorValue | undefined;
+  borderless?: null | boolean | undefined;
+  radius?: null | number | undefined;
+  foreground?: null | boolean | undefined;
+}
+
+export type PressEvent = {
+  changedTouches: PressEvent[];
+  identifier: number;
+  locationX: number;
+  locationY: number;
+  pageX: number;
+  pageY: number;
+  target: number;
+  timestamp: number;
+  touches: PressEvent[];
+  force?: number;
+};
+
+export type PressableEvent = { nativeEvent: PressEvent };
+
+export interface PressableProps
+  extends AccessibilityProps,
+    Omit<ViewProps, 'children' | 'style' | 'hitSlop'> {
+  /**
+   * Called when the hover is activated to provide visual feedback.
+   */
+  onHoverIn?: null | ((event: PressableEvent) => void);
+
+  /**
+   * Called when the hover is deactivated to undo visual feedback.
+   */
+  onHoverOut?: null | ((event: PressableEvent) => void);
+
+  /**
+   * Called when a single tap gesture is detected.
+   */
+  onPress?: null | ((event: PressableEvent) => void);
+
+  /**
+   * Called when a touch is engaged before `onPress`.
+   */
+  onPressIn?: null | ((event: PressableEvent) => void);
+
+  /**
+   * Called when a touch is released before `onPress`.
+   */
+  onPressOut?: null | ((event: PressableEvent) => void);
+
+  /**
+   * Called when a long-tap gesture is detected.
+   */
+  onLongPress?: null | ((event: PressableEvent) => void);
+
+  /**
+   * Either children or a render prop that receives a boolean reflecting whether
+   * the component is currently pressed.
+   */
+  children?:
+    | React.ReactNode
+    | ((state: PressableStateCallbackType) => React.ReactNode);
+
+  /**
+   * Whether a press gesture can be interrupted by a parent gesture such as a
+   * scroll event. Defaults to true.
+   */
+  cancelable?: null | boolean;
+
+  /**
+   * Duration to wait after hover in before calling `onHoverIn`.
+   * @platform macos windows
+   * | NOTE: not present in RN docs
+   */
```

Comment:
These props were present on the legacy `Pressable` but since they weren't documented anywhere, and I couldn't test them, I left that comment.

I'll remove them all together, I don't think we have to go that far with compatibility as we've already removed `onBlur` and `onFocus` due to lack of tools to implement those. (c.c. @j-piasecki )

Reply:
Sorry, I see that I actually implement those, but then couldn't test them :P
I think we can leave them in, I'll just have to find a way to build and test this on my computer.

Reply:
Still, I don't think it's a good idea to point out that supported platforms are desktops. Currently we have very limited support on `macOS` and we do not test things on `windows`. Also it sounds a bit weird that native library has desktop-only props, cc @j-piasecki.

Reply:
Alright, I think it's absolutely fair to remove those the same way `onBlur` and `onFocus` were removed.
I'll just wait for j-piasecki's opinion.

Reply:
They work on web, but I think we already talked about it - to implement them, we would have to track the mouse outside of the component, which doesn't align with gesture handlers capabilities.
And on native, it doesn't work even with mouse hooked up, and I tested this on a real device.

Reply:
> (...) we would have to track mouse clicks outside of the component, which doesn't align with gesture handlers capabilities

Those two props are called `delayHoverIn` and `delayHoverOut`, so correct me if I'm wrong, but doesn't it just start `hover` after delay? If so, why would we need to track mouse clicks outside of the component (also, in case of `hover` there's no click, unless I'm missing something)?

Reply:
Yes, hover tracking is currently fully functional, I was talking about `onBlur` and `onFocus` which activate on mouse click outside and inside of the component.

Back to the `delayHoverIn`, as far as I see in the code, it should be working, but I'll recheck on Monday.